A few tweaks for Python 3

msabramo opened the pull request:
Working on getting MySQLdb to work in Python 3.

Current status:

```
$ tox -e py33
...
Ran 69 tests in 6.115s

FAILED (errors=22)
...
```

methane commented on tox.ini:
```diff
@@ -0,0 +1,12 @@
+[tox]
+envlist = py25, py26, py27, py33
```
This pull request should not work with py25.

methane commented on MySQLdb/cursors.py:
```diff
@@ -14,6 +14,7 @@
     ListType = list
     TupleType = tuple
     UnicodeType = str
+    unicode = str
```
remove UnicodeType instead.

msabramo replied:
Why remove `UnicodeType`?

farcepest replied:
In Python 2.x, unicode is UnicodeType.  In Python 3.x, unicode (and UnicodeType) are set to str. UnicodeType is used at line 299. unicode is used in a variety of places.

methane commented on setup_posix.py:
```diff
@@ -1,5 +1,6 @@
 import os, sys
-from ConfigParser import SafeConfigParser
+import six.moves
+SafeConfigParser = six.moves.configparser.ConfigParser
```
Do you need six for only this?

methane commented on MySQLdb/connections.py:
```diff
@@ -229,8 +232,8 @@ def string_decoder(s):
-        self.encoders[types.StringType] = string_literal
-        self.encoders[types.UnicodeType] = unicode_literal
+        self.encoders[bytes] = string_literal
+        self.encoders[str] = unicode_literal
```
Add `unicode = str` like cursor and use it.

okin committed:
Improved support for Python 3.

okin committed:
Changing the way checks for the error code are made.

vstinner commented:
What is the status of this issue? I would like Python 3 support for MySQL-python.

msabramo committed:
tox.ini: Remove ipdb

It makes tests take longer to install deps and now causes this error on
Python < 2.7, because of IPython 2.0.0 requiring 2.7:

    ERROR: IPython requires Python Version 2.7 or above.

msabramo committed:
Remove MySQLdb/release.py from git

because it's generated from metadata.cfg

and because I keep getting annoying diffs because of newline differences
-- e.g.:

    $ git diff
    diff --git a/MySQLdb/release.py b/MySQLdb/release.py
    index 5c30a6c..4ce9412 100644
    --- a/MySQLdb/release.py
    +++ b/MySQLdb/release.py
    @@ -1,4 +1,4 @@
    -
    -__author__ = "Andy Dustman <farcepest@gmail.com>"
    -version_info = (1,2,4,'final',1)
    -__version__ = "1.2.4"
    +
    +__author__ = "Andy Dustman <farcepest@gmail.com>"
    +version_info = (1,2,4,'final',1)
    +__version__ = "1.2.4"

msabramo committed:
Support Python 2.7, 3.3, 3.4

These are reasonably modern versions of Python that I think are worth
supporting for the planned 1.3.0 release.

msabramo commented:
Note: https://github.com/farcepest/MySQLdb1/pull/48 is extremely similar.

methane committed:
Some Python 3 C-API fixes.

methane committed:
Fix memory corruption on Python 3.

methane committed:
Fix getattr on result object.

methane committed:
Fix segfault on Python 2.

methane committed:
Cleanup

Remove macros for Python <2.6

methane committed:
Remove more macros for ancient Python.

msabramo committed:
Merge branch 'py3-capi' into py3.3support

msabramo committed:
Merge branch 'support_py27_to_py34' into py3.3support

msabramo committed:
Merge branch 'remove_MySQLdb/release.py' into py3.3support

msabramo committed:
Handle types in a Python 3 friendly way

msabramo committed:
.travis.yml: Add py33 and py34 but allow fail

methane committed:
Fix TyepError on executemany.

methane committed:
Fix raw bytes literal for Python 2.

msabramo commented:
Check out #62. @methane has taken what I have here and added a bunch of great stuff, including modifying the C code to not segfault on Python 3.

msabramo committed:
Merge pull request #1 from methane/fix-typeerror-for-regex

Fix TyepError on executemany.

msabramo commented:
Ping

msabramo commented:
This project hasn't had a commit in over a year, so I'll just close this.
